activerecord/lib/active_record/associations.rb

```diff
@@ -1647,6 +1656,9 @@ def has_one(name, scope = nil, **options)
         #   be initialized with a particular record before validation.
         # [:strict_loading]
         #   Enforces strict loading every time the associated record is loaded through this association.
+        # [:owner_ensuring_destroy]
+        #   Specifies a instance method name that will be called in the owner when executing the background job to
+        #   delete the associated record to determine if the record should be deleted.
```

Comment:
Same update: `Specifies an instance method that will be called by the owner to destroy the associated record. The associated records will be destroyed in a background job`